src/js/panels.js

```diff
@@ -198,16 +193,317 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+                funcName: "gpii.firstDiscovery.panel.lang.overrideDefaultScroll",
+                args: ["{that}"]
+            },
+            "afterRender.scrollLangIntoView": {
+                funcName: "gpii.firstDiscovery.panel.lang.scrollLangIntoView",
+                args: ["{that}"]
+            },
+            "onButtonTopsReady.scrollLangIntoView": {
+                funcName: "gpii.firstDiscovery.panel.lang.scrollLangIntoView",
+                args: ["{that}"]
+            },
+            "afterRender.preventWrapWithArrowKeys": {
+                funcName: "gpii.firstDiscovery.panel.lang.preventWrapWithArrowKeys",
+                args: ["{that}"]
+            }
+        }
+    });
+
+    // This component is needed for the following demands block to be only applied to the language panel "gpii.firstDiscovery.panel.lang".
+    // Without this component being the sub-component of the language panel, according to http://wiki.fluidproject.org/display/docs/Contexts,
+    // when the context component of the demands block was the language panel itself, the demands block would also be applied to siblings of
+    // the language panel. To work around this issue, another layer of containment needs to be added.
+    // This component and the demands block should be removed when the new framework (http://issues.fluidproject.org/browse/FLUID-5249)
+    // is in use.
+    fluid.defaults("gpii.firstDiscovery.panel.lang.attachTooltipOnLang", {
+        gradeNames: ["gpii.firstDiscovery.attachTooltip", "autoInit"],
+        tooltipContentMap: {
+            "prev": "navButtonTooltip",
+            "next": "navButtonTooltip"
+        }
+    });
+
+    fluid.demands("fluid.tooltip", ["gpii.firstDiscovery.panel.lang.attachTooltipOnLang"], {
+        options: {
+            styles: {
+                tooltip: "gpii-fd-tooltip-lang"
+            }
+        }
+    });
+
+    gpii.firstDiscovery.panel.lang.setNavKeyStatus = function (that) {
+        var langArray = that.options.controlValues.lang,
+            selectedLang = that.model.lang;
+
+        that.locate("prev").prop("disabled", selectedLang === langArray[0]);
+        that.locate("next").prop("disabled", selectedLang === langArray[langArray.length - 1]);
+    };
+
+    gpii.firstDiscovery.panel.lang.moveLangFocus = function (that, adjustBy) {
+        var langArray = that.options.controlValues.lang,
+            nextIndex = langArray.indexOf(that.model.lang) + adjustBy;
+
+        if (nextIndex >= 0 && nextIndex <= langArray.length) {
+            that.applier.change("lang", langArray[nextIndex]);
+        }
+    };
+
+    // find the index of the number in the "numbers" array that's closest to the given "currentNumber"
+    gpii.firstDiscovery.panel.lang.findClosestNumber = function (currentNumber, numbers) {
+        var distance = Math.abs(numbers[0] - currentNumber),
+            idx = 0;
+
+        for (var c = 1; c < numbers.length; c++) {
```

Comment:
A more regular loop will start from 0 and initialise distance at Infinity, idx at -1